pkg/registry/apps/shorturl/legacy_storage.go

```diff
@@ -199,3 +207,29 @@ func (s *legacyStorage) Delete(ctx context.Context, name string, deleteValidatio
 func (s *legacyStorage) DeleteCollection(ctx context.Context, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions, listOptions *internalversion.ListOptions) (runtime.Object, error) {
 	return nil, fmt.Errorf("DeleteCollection for shorturl not implemented")
 }
+
+// convertRequesterToSignedInUser converts any identity.Requester to *user.SignedInUser
+// This is needed because some legacy shorturls service methods still expect SignedInUser
+func convertRequesterToSignedInUser(requester identity.Requester) (*user.SignedInUser, error) {
+	// If it's already a SignedInUser, return it directly
+	if signedInUser, ok := requester.(*user.SignedInUser); ok {
+		return signedInUser, nil
+	}
+
+	// If it's an authn.Identity, use its SignedInUser method
+	if authnIdentity, ok := requester.(*authn.Identity); ok {
+		return authnIdentity.SignedInUser(), nil
+	}
+
+	// For all other identity types (background services, etc.), create a simple background user
+	orgID := requester.GetOrgID()
+	if orgID <= 0 {
+		orgID = 1 // Default to org 1 for system operations
+	}
+
+	return &user.SignedInUser{
+		OrgID:   orgID,
+		OrgRole: org.RoleAdmin, // Background services need admin access for cleanup operations
+		Login:   "grafana_background_service",
+	}, nil
```

Comment:
Rather than have this as the default, could you add it to the background operation context instead?

Here's an example: https://github.com/grafana/grafana/blob/main/pkg/services/dashboards/service/dashboard_service.go#L198